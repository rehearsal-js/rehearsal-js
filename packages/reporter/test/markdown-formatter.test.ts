import { resolve } from 'node:path';
import { readJSONSync } from 'fs-extra/esm';
import { describe, expect, test } from 'vitest';
import { MarkdownFormatter } from '../src/formatters/index.js';

import type { Report } from '../src/types.js';

const rehearsalReportJSON = readJSONSync(
  resolve('./test/fixtures/rehearsal-report.json')
) as Report;

describe('MarkdownFormatter', () => {
  test('should transform all fields correctly, irregardless of undefined, missing values', () => {
    expect(MarkdownFormatter.getReport(rehearsalReportJSON)).matchSnapshot();
    expect(MarkdownFormatter.extension).toBe('.md');
  });
});
