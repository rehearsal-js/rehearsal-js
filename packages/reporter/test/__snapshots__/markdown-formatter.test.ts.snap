// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MarkdownFormatter > should transform all fields correctly, irregardless of undefined, missing values 1`] = `
"### Summary:
Project Name: @rehearsal/test
Typescript Version: 4.7.4
Report path: undefined
timestamp: 9/16/2022, 13:24:57

### Results:

#### File: /base/path/file1.ts, issues: 3:

**Error TS6133**: 'REVIEW REQUIRED'
The declaration 'react' is never read or used. Remove the declaration or use it.
Code: \`import react from 'react';\`

**Error TS2322**: 'REVIEW REQUIRED'
The variable 'a' has type 'number', but 'string' is assigned. Please convert 'string' to 'number' or change variable's type.
Code: \`a\`

**Error TS2345**: 'REVIEW REQUIRED'
Argument of type '{0}' is not assignable to parameter of type 'string'. Consider verifying both types, using type assertion: '({} as string)', or using type guard: 'if ({} instanceof string) { ... }'.
Code: \`{}\`

#### File: /base/path/file2.ts, issues: 1:

**Error @typescript-eslint/no-explicit-any**: 'REVIEW REQUIRED'
Unexpected any. Specify a different type.
Code: \`\`
"
`;
